--- TS/Slang.lean (before this round)
+++ TS/Slang.lean
@@ -294,5 +294,5 @@
 
 Before applying `fun_induction` to a function as complex as {name}`Aexp.optimize0plus`,
-let's see how it works on somthing simpler. Recall the definition of `Nat.even` and `Nat.odd`:
+let's see how it works on something simpler. Recall the definition of `Nat.even` and `Nat.odd`:
 
 ```lean
@@ -306,7 +306,7 @@
 ```
 
-Normally, if we perform induction on `n`, we get two cases - `0` and `n' + 1` -
+Normally, if we perform induction on `n`, we get two cases -- `0` and `n' + 1` --
 one for each of the cases in the inductive definition of natural numbers.
-Functional induction on {name}`Nat.even`, however, gives us three cases - `0`, `1`, and `n' + 2` -
+Functional induction on {name}`Nat.even`, however, gives us three cases -- `0`, `1`, and `n' + 2` --
 corresponding to each of the cases of its definition.
 
@@ -573,5 +573,5 @@
 About `Bexp.eval` below: We should discuss a way to recall definitions without
 having to write them out manually like this. I think a simple `#print` may work as an
-alternative, assuming there are no namespace issues..
+alternative, assuming there are no namespace issues.
 :::
 
@@ -881,5 +881,5 @@
 
 Where the thing being defined is not easy to express as a function,
-definitions are often simpler. When both
+relational definitions are often simpler. When both
 styles are workable, relational definitions can be more elegant and
 easier to understand, and Lean generates useful inversion and induction
